pepper-apis/dsm-core/src/test/java/org/broadinstitute/dsm/service/adminoperation/LegacyKitResampleServiceTest.java

```diff
@@ -0,0 +1,222 @@
+        ElasticTestUtil.createParticipant(esIndex, participantDto);
+        participants.add(participantDto);
+        ddpInstance = DDPInstance.getDDPInstance(instanceName);
+        String dsmKitRequestId = testKitUtil.createKitRequestShipping(ddpParticipantId, oldCollaboratorSampleId,
+                oldCollaboratorParticipantId, null,  ddpKitRequestId, "SALIVA", ddpInstance, "100");
+        createdKits.add(dsmKitRequestId);
+    }
+
+    @AfterClass
+    public static void tearDown() {
+        participants.forEach(participantDto ->
+                TestParticipantUtil.deleteParticipant(participantDto.getRequiredParticipantId()));
+        createdKits.forEach(kitRequestId -> testKitUtil.deleteKitRequestShipping(Integer.parseInt(kitRequestId)));
+        testKitUtil.deleteGeneratedData();
+        ElasticTestUtil.deleteIndex(esIndex);
+    }
+
+    @After
+    public void deleteParticipantData() {
+        MedicalRecordTestUtil.deleteInstanceMedicalRecordBundles(ddpInstanceDto);
+        TestParticipantUtil.deleteInstanceParticipants(ddpInstanceDto);
+    }
+
+    @Test
+    public void testVerify_notVerifyMissingOldCollaboratorSampleId() {
+        LegacyKitResampleRequest wrongKitResampleRequest = new LegacyKitResampleRequest(null,
+                newCollaboratorSampleId, newCollaboratorParticipantId, shortId, legacyShortId);
+        try {
+            wrongKitResampleRequest.verify(ddpInstance, new KitRequestDao());
+        } catch (Exception e) {
+            e.printStackTrace();
+            Assert.assertEquals("Missing required field: currentCollaboratorSampleId", e.getMessage());
+            return;
+        }
+        Assert.fail("Should have thrown exception");
+    }
+
+    @Test
+    public void testVerify_notVerifyMissingNewCollaboratorSampleId() {
+        LegacyKitResampleRequest wrongKitResampleRequest = new LegacyKitResampleRequest(oldCollaboratorSampleId,
+                null, newCollaboratorParticipantId, shortId, legacyShortId);
+        try {
+            wrongKitResampleRequest.verify(ddpInstance, new KitRequestDao());
+        } catch (Exception e) {
+            e.printStackTrace();
+            Assert.assertEquals("Missing required field: newCollaboratorSampleId", e.getMessage());
+            return;
+        }
+        Assert.fail("Should have thrown exception");
+    }
+
+    @Test
+    public void testVerify_notVerifyMissingShortId() {
+        LegacyKitResampleRequest wrongKitResampleRequest = new LegacyKitResampleRequest(oldCollaboratorSampleId,
+                newCollaboratorSampleId, newCollaboratorParticipantId, null, legacyShortId);
+        try {
+            wrongKitResampleRequest.verify(ddpInstance, new KitRequestDao());
+        } catch (Exception e) {
+            e.printStackTrace();
+            Assert.assertEquals("Missing required field: shortId", e.getMessage());
+            return;
+        }
+        Assert.fail("Should have thrown exception");
+    }
+
+    @Test
```

Comment:
I'm not sure a separate test for missing each of the possible request field adds much value, since the underlying mechanism for each is the same.